pkg/apis/core/helper/helpers.go

```diff
@@ -12,21 +12,20 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package componentconfig
+package helper
 
 import (
-	"os"
+	"github.com/gardener/gardener/pkg/apis/core"
 )
 
-// ApplyEnvironmentToConfig checks for several well-defined environment variables and if they are set,
-// it sets the value of the respective keys of <config> to the values in the environment.
-// Currently implemented environment variables:
-// KUBECONFIG can override config.ClientConnection.KubeConfigFile
-func ApplyEnvironmentToConfig(config *ControllerManagerConfiguration) {
-	if kubeconfig := os.Getenv("KUBECONFIG"); kubeconfig != "" {
-		config.ClientConnection.KubeConfigFile = kubeconfig
-	}
-	if kubeconfig := os.Getenv("GARDENER_KUBECONFIG"); kubeconfig != "" {
-		config.GardenerClientConnection.KubeConfigFile = kubeconfig
+// GetCondition returns the condition with the given <conditionType> out of the list of <conditions>.
+// In case the required type could not be found, it returns nil.
+func GetCondition(conditions []core.Condition, conditionType core.ConditionType) *core.Condition {
```

Comment:
Can't we somehow circumvent writing this stuff for the internal and external versions? Same goes for the validations - Reusing parts of them externally is really hard...

Reply:
I don't think we can, or do you have a concrete idea? The internal types might look completely different - it's more or less a special case right now that they are equal.

Reply:
Nah this was just me thinking out loud. I also see that the types might be vastly different.
I guess this eventually boils down to the strategy how you want to work with your types - Do our controllers always act upon the internal version (and thus require the helpers on these versions) or do we keep everything somehow compatible and only act upon the newest version.